<?xml version="1.0" encoding="utf-8"?>
<!--
Launch file for Selective ICP with rviz
-->

<launch>

  <!-- Proper Output Info -->
  <env name="ROSCONSOLE_FORMAT" value="[${severity}] (${node}) [${time}]: ${message}\n" />

  <param name="/use_sim_time" value="true" />
    <!--<arg name="cad_file" default="$(find cpt_selective_icp)/resources/cla.json" />-->
    <arg name="rosbag" />
    <arg name="logprefix" />
    <arg name="segmentation_model" />
    <arg name="input_height" />
    <arg name="input_width" />
    <arg name="threshold" default="150" />

    <node pkg="background_foreground_segmentation"
          type="inference_node2.py"
          name="inference_node"
          output="screen">
        <param name="model_gdrive_id" value="$(arg segmentation_model)" />
        <param name="input_height" value="$(arg input_height)" />
        <param name="input_width" value="$(arg input_width)" />
        <param name="pseudo_rgb" value="true" />
        <rosparam>
          image_topics:
            - /helios_left/image
            - /helios_right/image
          segmentation_output_topics:
            - /helios_left/segmented
            - /helios_right/segmented
        </rosparam>
    </node>

    <node pkg="background_foreground_segmentation"
          type="depth_filtering_node2"
          name="depth_filter_left"
          output="screen">
        <param name="density_threshold" value="$(arg threshold)" />
        <rosparam param="cloud_topics">
            - /helios_left/points_raw
        </rosparam>
        <rosparam param="caminfo_topics">
            - /helios_left/camera_info
        </rosparam>
        <rosparam param='mask_topics'>
            - /helios_left/segmented
        </rosparam>
    </node>

    <node pkg="background_foreground_segmentation"
          type="depth_filtering_node2"
          name="depth_filter_right"
          output="screen">
        <param name="density_threshold" value="$(arg threshold)" />
        <rosparam param="cloud_topics">
            - /helios_right/points_raw
        </rosparam>
        <rosparam param="caminfo_topics">
            - /helios_right/camera_info
        </rosparam>
        <rosparam param='mask_topics'>
            - /helios_right/segmented
        </rosparam>
    </node>

    <!---->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find background_foreground_segmentation)/config/tof.rviz" />
    <!-- should be started last for handover to ICP-->
    <include file="$(find background_foreground_segmentation)/launch/tof_tools/tof_from_bag.launch">
        <arg name="rosbag" value="$(arg rosbag)" />
    </include>

</launch>
